internal/database/webhooks.go

```diff
@@ -150,20 +151,56 @@ type WebhookNotFoundError struct {
 	ID uuid.UUID
 }
 
-func (w *WebhookNotFoundError) Error() string {
+func (w WebhookNotFoundError) Error() string {
 	return fmt.Sprintf("webhook with UUID %s not found", w.ID)
 }
 
-func (w *WebhookNotFoundError) NotFound() bool {
+func (w WebhookNotFoundError) NotFound() bool {
 	return true
 }
 
 // Update the webhook
 func (s *webhookStore) Update(ctx context.Context, newWebhook *types.Webhook) (*types.Webhook, error) {
-	// TODO(sashaostrikov) implement this method
-	panic("implement this method")
+	var (
+		err             error
+		encryptedSecret string
+		keyID           string
+	)
+	if newWebhook.Secret != nil {
+		encryptedSecret, keyID, err = newWebhook.Secret.Encrypt(ctx, s.key)
+		if err != nil || (encryptedSecret == "" && keyID == "") {
+			return nil, errors.Wrap(err, "encrypting secret")
+		}
+	}
+
+	q := sqlf.Sprintf(webhookUpdateQueryFmtstr,
+		newWebhook.CodeHostURN, encryptedSecret, newWebhook.ID, newWebhook.UUID,
+		sqlf.Join(webhookColumns, ", "))
+
+	updated, err := scanWebhook(s.QueryRow(ctx, q), s.key)
+	if err != nil {
+		if strings.Contains(err.Error(), "no rows in result set") {
```

Comment:
I think matching on the exact type is safer here:

```suggestion
		if err == sql.ErrNoRows {
```